tools/topology/CMakeLists.txt

```diff
@@ -6,6 +6,14 @@ file(GLOB TPLG_DEPS
 	sof/*.m4
 )
 
+add_custom_target(abi
+	COMMAND ./get_abi.sh
+	DEPENDS ${TPLG_DEPS}
+	WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
```

Comment:
You shouldn't output anything to source directories. Everything you produce should be put in build tree. Please pass here argument for output dir or just change workdir to output dir if you don't want to change `get_abi.sh`. To make `include('abi.h')` work, just add -I to m4 to something like `<builddir>/generated`